beacon_node/store/src/hot_cold_store.rs

```diff
@@ -1142,6 +1144,14 @@ impl<E: EthSpec, Hot: ItemStore<E>, Cold: ItemStore<E>> HotColdDB<E, Hot, Cold>
         }
     }
 
+    /// Rebase state pubkey_cache on some existing state to prevent re-populating the cache on the
+    /// first process_deposit call.
+    fn rebase_caches(&self, state: &mut BeaconState<E>) {
+        if let Ok((_, first_state)) = self.state_cache.lock().iter().exactly_one() {
+            state.rebase_caches_on(first_state);
+        }
+    }
```

Comment:
@michaelsproul is it possible to access the head state here? 
(2) Those the state_cache always include the head state? 
(3) Is the state_cache guaranteed to never be empty?

Reply:
Oh, on `unstable` that `state_cache` is just for historic states. It would be better named `historic_state_cache`.

The `state_cache` on `tree-states` _can_ guarantee that the **finalized** state is always present. See:

https://github.com/sigp/lighthouse/blob/928915c71885a2439f8fa3b3a22ef6b39e978339/beacon_node/store/src/state_cache.rs#L35

(the `Option` is just for first-time initialization)

In summary:

1) No, but you could get the `finalized` state on `tree-states`. I think that would work?
2) Definitely not on unstable (head is not historic), and _probably yes_ (but not guaranteed) on `tree-states`.
3) On unstable, no; on tree-states, yes.

Reply:
Might be best to postpone this PR for post tree-states then